### PR TITLE
Orca v0.1 - C + Python

### PR DESCRIPTION
**Pull request by bejager**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,208 @@
...
+### Audio output
+
+Orca's synthesized speech is delivered in a single-channel 16-bit PCM format and can be directly saved to a WAV file.
+The output can be directly fed into a playback audio system.
+
+## AccessKey
+
+AccessKey is your authentication and authorization token for deploying Picovoice SDKs, including Orca. Anyone who is
+using Picovoice needs to have a valid AccessKey. You must keep your AccessKey secret. You would need internet
+connectivity to validate your AccessKey with Picovoice license servers even though the speaker recognition is running
```
Comment (mrrostam) [inline README.md:83]: speaker recognition?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,208 @@
...
+### C
+
+The header file [include/pv_orca.h](./include/pv_orca.h) contains relevant information on Orca's C SDK.
+
+Build an instance of the object:
+
+```c
+pv_orca_t *handle = NULL;
+const char *model_path = "${MODEL_PATH}";
+pv_status_t status = pv_orca_init(${ACCESS_KEY}, model_path, &handle);
```
Comment (mrrostam) [inline README.md:160]:
```suggestion
pv_status_t status = pv_orca_init("${ACCESS_KEY}", model_path, &handle);
```

File: resources/.lint/spell-check/dict.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+Diarizing
```
Comment (mrrostam) [inline resources/.lint/spell-check/dict.txt:1]: some of these seem irrelevant to Orca

File: resources/.lint/spell-check/.cspell.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    "**/gradle-wrapper.properties",
+    "**/gradle.properties",
+    "**/gradlew",
+    "**/gradlew.bat",
+    "**/proguard-rules.pro",
+
+    // android
+    "**/AndroidManifest.xml",
+    "**/activity_main.xml",
+    "**/koala_activity_demo.xml"
```
Comment (mrrostam) [inline resources/.lint/spell-check/.cspell.json:51]:
```suggestion
    "**/orca_activity_demo.xml"
```

File: resources/.lint/spell-check/.cspell.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+  "ignorePaths": [
+    // binaries
+    "**/*.a",
+    "**/*.dll",
+    "**/*.dylib",
+    "**/*.jar",
+    "**/*.mp3",
+    "**/*.node",
+    "**/*.ppn",
```
Comment (mrrostam) [inline resources/.lint/spell-check/.cspell.json:21]: ?

File: demo/python/setup.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    author_email="hello@picovoice.ai",
+    description="Orca Text-to-Speech Engine demos",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/Picovoice/orca",
+    packages=["pvorcademo"],
+    install_requires=["pvorca==0.1.0"],
+    include_package_data=True,
+    classifiers=[
+        "Development Status :: 5 - Production/Stable",
```
Comment (mrrostam) [inline demo/python/setup.py:38]:
```suggestion
        "Development Status :: 4 - Beta",
```

File: demo/python/setup.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        "Operating System :: OS Independent",
+        "Programming Language :: Python :: 3",
+        "Topic :: Multimedia :: Sound/Audio :: Speech"
+    ],
+    entry_points=dict(
+        console_scripts=[
+            'orca_demo=pvorcademo.orca_demo:main',
+        ],
+    ),
+    python_requires='>=3.5',
```
Comment (mrrostam) [inline demo/python/setup.py:50]: let's move the min version of python to 3.7+
https://devguide.python.org/versions/

Comment (mrrostam) [inline demo/python/setup.py:50]:
```suggestion
    python_requires='>=3.7',
```

File: demo/python/orca_demo.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        length_sec = len(pcm) / orca.sample_rate
+        with wave.open(args.output_path, 'wb') as output_file:
+            output_file.setnchannels(1)
+            output_file.setsampwidth(2)
+            output_file.setframerate(orca.sample_rate)
+            output_file.writeframes(struct.pack('%dh' % len(pcm), *pcm))
+    except OrcaActivationLimitError:
+        print('AccessKey has reached its processing limit')
+    finally:
+        print('%.2f seconds of audio were written to `%s`.' % (length_sec, args.output_path))
```
Comment (mrrostam) [inline demo/python/orca_demo.py:59]: it seems this should be moved to the `try` block

File: demo/python/README.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+## Orca
+
+Orca is an on-device text-to-speech engine producing high-quality, realistic, spoken audio with zero latency. Orca is:
+
+- Private; All voice processing runs locally.
+- Cross-Platform:
+    - Linux (x86_64), macOS (x86_64, arm64), Windows (x86_64)
+    - Android and iOS
+    - Chrome, Safari, Firefox, and Edge
+    - Raspberry Pi (4, 3) and NVIDIA Jetson Nano
```
Comment (mrrostam) [inline demo/python/README.md:14]:
```suggestion
    - Raspberry Pi (5, 4, 3) and NVIDIA Jetson Nano
```

File: demo/python/README.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+## Compatibility
+
+- Python 3.5+
+- Runs on Linux (x86_64), macOS (x86_64, arm64), Windows (x86_64), Raspberry Pi (4, 3), and NVIDIA Jetson Nano.
```
Comment (mrrostam) [inline demo/python/README.md:19]:
```suggestion
- Runs on Linux (x86_64), macOS (x86_64, arm64), Windows (x86_64), Raspberry Pi (5, 4, 3), and NVIDIA Jetson Nano.
```

File: demo/python/README.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+- Private; All voice processing runs locally.
+- Cross-Platform:
+    - Linux (x86_64), macOS (x86_64, arm64), Windows (x86_64)
+    - Android and iOS
+    - Chrome, Safari, Firefox, and Edge
+    - Raspberry Pi (4, 3) and NVIDIA Jetson Nano
+
+## Compatibility
+
+- Python 3.5+
```
Comment (mrrostam) [inline demo/python/README.md:18]:
```suggestion
- Python 3.7+
```

File: demo/c/orca_demo.c [hunk 12]
```diff
@@ -0,0 +1,381 @@
+/*
+Copyright 2019-2024 Picovoice Inc.
```
Comment (mrrostam) [inline demo/c/orca_demo.c:2]:
```suggestion
Copyright 2024 Picovoice Inc.
```

File: binding/python/setup.py [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    author="Picovoice",
+    author_email="hello@picovoice.ai",
+    description="Orca Text-to-Speech Engine.",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    url="https://github.com/Picovoice/orca",
+    packages=["pvorca"],
+    include_package_data=True,
+    classifiers=[
+        "Development Status :: 5 - Production/Stable",
```
Comment (mrrostam) [inline binding/python/setup.py:55]:
```suggestion
        "Development Status :: 4 - Beta",
```

File: binding/python/setup.py [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: Apache Software License",
+        "Operating System :: OS Independent",
+        "Programming Language :: Python :: 3",
+        "Topic :: Multimedia :: Sound/Audio :: Speech",
+    ],
+    python_requires='>=3.5',
```
Comment (mrrostam) [inline binding/python/setup.py:62]:
```suggestion
    python_requires='>=3.7',
```

File: binding/python/_util.py [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+        cpu_part = cpu_part_list[0].split(" ")[-1].lower()
+    except Exception as e:
+        raise RuntimeError("Failed to identify the CPU with `%s`\nCPU info: `%s`" % (e, cpu_info))
+
+    if "0xd03" == cpu_part:
+        return "cortex-a53" + arch_info
+    elif "0xd07" == cpu_part:
+        return "cortex-a57" + arch_info
+    elif "0xd08" == cpu_part:
+        return "cortex-a72" + arch_info
```
Comment (mrrostam) [inline binding/python/_util.py:43]: rpi5?

Comment (bejager) [reply binding/python/_util.py:43]: Added!

File: binding/python/_util.py [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+    elif "0xd0b" == cpu_part:
+        return "cortex-a76" + arch_info
+    else:
+        raise NotImplementedError("Unsupported CPU: `%s`." % cpu_part)
+
+
+_RASPBERRY_PI_MACHINES = {"cortex-a53", "cortex-a72", "cortex-a53-aarch64", "cortex-a72-aarch64"}
```
Comment (mrrostam) [inline binding/python/_util.py:50]: You also need to include RPI5 here; @laves  I believe we should set up the RPI5 runner before merging this. btw, rpi5 also should be added to the action scripts

File: binding/python/test_orca.py [hunk 17]
```diff
@@ -0,0 +1,135 @@
...
+#
+#    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+#    file accompanying this source.
+#
+#    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+#    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+#    specific language governing permissions and limitations under the License.
+#
+
+import argparse
```
Comment (mrrostam) [inline binding/python/test_orca.py:12]: so we dont have any test to check the generated pcm, right? Can we use Leopard to calculate WER/CER on it?

File: binding/python/README.md [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+
+- `speech_rate`: This controls the speed of the generated speech. A higher (lower) value produced
+speech that is faster (slower). The default is `1.0`. 
+
+When done make sure to explicitly release the resources with `orca.delete()`.
+
+
+## Demos
+
+pvorcademo [TODO: Add pip link after release] provides command-line utilities for processing audio using Orca.
```
Comment (laves) [inline binding/python/README.md:78]:
```suggestion
pvorcademo [TODO: Add pip link after release] provides command-line utilities for synthesizing audio using Orca.
```

File: binding/python/_orca.py [hunk 19]
```diff
@@ -0,0 +1,396 @@
...
+
+        :param text: Text to be converted to audio. The maximum number of characters per call to `.synthesize()` is
+        `.max_character_limit()`. Allowed characters are lower-case and upper-case letters and punctuation marks
+        that can be retrieved with `.valid_punctuation_symbols()`.
+        Custom pronunciations can be embedded in the text via the syntax `{word|pronunciation}`.
+        The pronunciation is expressed in ARPAbet format, e.g.: "I {live|L IH V} in {Sevilla|S EH V IY Y AH}".
+        :param output_path: Absolute path to the output audio file. The output file is saved as `WAV (.wav)`
+        and consists of a single mono channel.
+        :param speech_rate: Rate of speech of the generated audio.
+        :return: True if the operation is successful, throws an exception otherwise.
```
Comment (laves) [inline binding/python/_orca.py:325]: if it's either true or exception, it sounds like this function should just be void because you essentially know that it succeeded if it returned successfully

File: binding/python/_util.py [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+                os.path.dirname(__file__), relative, "lib/raspberry-pi/%s/libpv_orca.so" % linux_machine
+            )
+    elif platform.system() == "Windows":
+        return os.path.join(os.path.dirname(__file__), relative, "lib", "windows", "amd64", "libpv_orca.dll")
+
+    raise NotImplementedError("Unsupported platform.")
+
+
+def default_model_path(relative: str = "") -> str:
+    return os.path.join(os.path.dirname(__file__), relative, "lib", "common", "orca_params_f1.pv")
```
Comment (laves) [inline binding/python/_util.py:77]: so I guess we're going to ship one default voice with the package and then they have to go to github if they want other voices?

File: binding/python/setup.py [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+for rel_path in INCLUDE_FILES:
+    shutil.copy(os.path.join(os.path.dirname(__file__), rel_path), package_folder)
+    manifest_in += "include pvorca/%s\n" % os.path.basename(rel_path)
+
+os.mkdir(os.path.join(package_folder, 'lib'))
+for platform in INCLUDE_LIBS:
+    shutil.copytree(
+        os.path.join(os.path.dirname(__file__), '../../lib', platform),
+        os.path.join(package_folder, 'lib', platform))
+manifest_in += "recursive-include pvorca/lib *\n"
```
Comment (laves) [inline binding/python/setup.py:35]: correct me if I'm wrong but this will include both the male and female models, but above it looks like only the female model is to the user? If this is the case, we should only ship one model or build in the ability for the user to select between male and female default models somehow

File: demo/c/test/test_orca_c.py [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+            "-o", output_path,
+        ]
+
+        process = subprocess.Popen(args, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+
+        self.assertEqual(process.poll(), 0)
+        self.assertEqual(stderr.decode('utf-8'), '')
+        self.assertTrue("Saved audio" in stdout.decode('utf-8'))
+        os.remove(output_path)
```
Comment (laves) [inline demo/c/test/test_orca_c.py:71]: we should probably be checking output audio here as well. Though this is directly using the C interface, it still is a good test of whether the models or libs on github are generating unexpected results

**Review by laves — CHANGES_REQUESTED** [review]

File: binding/python/README.md [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+Made in Vancouver, Canada by [Picovoice](https://picovoice.ai)
+
+Orca is an on-device text-to-speech engine producing high-quality, realistic, spoken audio with zero latency. Orca is:
+
+- Private; All voice processing runs locally.
+- Cross-Platform:
+    - Linux (x86_64), macOS (x86_64, arm64), Windows (x86_64)
+    - Android and iOS
+    - Chrome, Safari, Firefox, and Edge
+    - Raspberry Pi (5, 4, 3) and NVIDIA Jetson Nano
```
Comment (ksyeo1010) [inline binding/python/README.md:14]: some say (5, 4, 3) some (3, 4, 5) I guess we can keep all the same.

File: binding/python/README.md [hunk 24]
```diff
@@ -0,0 +1,79 @@
...
+- "This is a {custom|K AH S T AH M} pronunciation"
+- "{read|R IY D} this as {read|R EH D}, please."
+- "I {live|L IH V} in {Sevilla|S EH V IY Y AH}. We have great {live|L AY V} sports!"
+
+Replace `${OUTPUT_PATH}` with the path to save the generated audio as a single-channel 16-bit PCM WAV file.\
+Retrieve the sample rate with `orca.sample_rate`.
+
+Optional keyword arguments for the `synthesize` methods:
+
+- `speech_rate`: This controls the speed of the generated speech. A higher (lower) value produced
```
Comment (ksyeo1010) [inline binding/python/README.md:70]:
```suggestion
- `speech_rate`: This controls the speed of the generated speech. A higher (lower) value produces
```

File: binding/python/_orca.py [hunk 25]
```diff
@@ -0,0 +1,396 @@
...
+    'Orca',
+    'OrcaActivationError',
+    'OrcaActivationLimitError',
+    'OrcaActivationRefusedError',
+    'OrcaActivationThrottledError',
+    'OrcaError',
+    'OrcaInvalidArgumentError',
+    'OrcaIOError',
+    'OrcaMemoryError',
+    'OrcaRuntimeError',
```
Comment (ksyeo1010) [inline binding/python/_orca.py:395]: pretty sure they are more errors in other sdks, is there reason we exlucded them?

File: demo/c/orca_demo.c [hunk 26]
```diff
@@ -0,0 +1,381 @@
...
+
+#endif
+}
+
+static struct option long_options[] = {
+        {"access_key",   required_argument, NULL, 'a'},
+        {"library_path", required_argument, NULL, 'l'},
+        {"model_path",   required_argument, NULL, 'm'},
+        {"text",         required_argument, NULL, 't'},
+        {"output_path",  required_argument, NULL, 'o'},
```
Comment (ksyeo1010) [inline demo/c/orca_demo.c:90]: I guess we are going with `wav_output_path`?

File: demo/python/orca_demo.py [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+    parser.add_argument(
+        '--access_key',
+        required=True,
+        help='AccessKey obtained from Picovoice Console (https://console.picovoice.ai/)')
+    parser.add_argument(
+        '--text',
+        required=True,
+        help='Text to be synthesized')
+    parser.add_argument(
+        '--output_path',
```
Comment (ksyeo1010) [inline demo/python/orca_demo.py:30]:
```suggestion
        '--wav_output_path',
```

File: binding/python/test_orca.py (hunk 17 above)
Comment (bejager) [reply binding/python/test_orca.py:12]: Added leopard to the python sdk tests

File: README.md [hunk 28]
```diff
@@ -1,2 +1,208 @@
...
+
+To synthesize speech with a specific voice, pass the corresponding model file to the orca init function.
+
+### Speech control
+
+Orca provides a set of parameters to control the synthesized speech. The following table lists the available parameters:
+
+|  Parameter  | Default |                                    Description                                    |
+|:-----------:|:-------:|:---------------------------------------------------------------------------------:|
+| speech rate |   1.0   | Speed of generated speech. Higher (lower) values generate faster (slower) speech. |
```
[the remark this answers is not in the document]
Comment (bejager) [reply README.md]: Yes, I added it to the READMEs

File: binding/python/_orca.py (hunk 19 above)
Comment (bejager) [reply binding/python/_orca.py]: Right, made it void!

File: binding/python/_util.py (hunk 20 above)
Comment (bejager) [reply binding/python/_util.py]: Good point. So the female one is set up as default. For python, I guess we decided to ship both models right away? And for the other SDKs we can stick to the female one and tell the user to download the others when needed.

File: binding/python/setup.py (hunk 21 above)
Comment (bejager) [reply binding/python/setup.py:35]: See above. It's handled via the init function. I will also add it to the READMEs.

File: binding/python/test_orca.py [hunk 29]
```diff
@@ -0,0 +1,135 @@
...
+        self.assertGreater(len(pcm), 0)
+
+        pcm_custom = self.orca.synthesize(self.TEXT_CUSTOM_PRON)
+        self.assertGreater(len(pcm_custom), 0)
+
+        output_path = os.path.join(os.path.dirname(__file__), "output.wav")
+        ok = self.orca.synthesize_to_file(self.TEXT, output_path=output_path)
+        self.assertTrue(ok)
+        self.assertTrue(os.path.isfile(output_path))
+        os.remove(output_path)
```
[the remark this answers is not in the document]
Comment (bejager) [reply binding/python/test_orca.py]: As discussed in scrum, this is a bit more tricky. For the Python SDK I added leopard, for the rest I added a card to tracking.

File: demo/c/test/test_orca_c.py (hunk 22 above)
Comment (bejager) [reply demo/c/test/test_orca_c.py:65]: See above

File: include/picovoice.h [hunk 30]
```diff
@@ -0,0 +1,85 @@
...
+#define PV_API __attribute__((visibility("default")))
+
+    /**
+* Audio sample rate accepted by Picovoice.
+*/
+    PV_API int32_t pv_sample_rate(void);
+
+    /**
+* Status codes.
+*/
```
[the remark this answers is not in the document]
Comment (bejager) [reply include/picovoice.h]: Fixed!

File: include/pv_orca.h [hunk 31]
```diff
@@ -0,0 +1,205 @@
...
+PV_API void pv_orca_valid_punctuation_symbols_delete(const char *const *symbols);
+
+/**
+ * Gets the sampling rate of the audio produced by Orca.
+ *
+ * @param object Constructed instance of Orca.
+ * @param[out] sample_rate Sampling rate of the audio produced by Orca.
+ * @return Status code. Returns `PV_STATUS_INVALID_ARGUMENT` on failure.
+ */
+PV_API pv_status_t pv_orca_sample_rate(const pv_orca_t *object, int32_t *sample_rate);
```
[the remark this answers is not in the document]
Comment (bejager) [reply include/pv_orca.h:85]: Removed the sample_rate function from the picovoice header.

File: binding/python/_orca.py (hunk 25 above)
Comment (bejager) [reply binding/python/_orca.py]: Good catch, they were missing in the export but have been defined at the top of the file!

File: demo/python/orca_demo.py (hunk 27 above)
Comment (bejager) [reply demo/python/orca_demo.py:30]: See above

File: demo/c/orca_demo.c (hunk 26 above)
Comment (bejager) [reply demo/c/orca_demo.c:90]: I was thinking of leaving this more general? We might add more output options at some point.

**Comment by bejager** [comment]
Thanks for the feedback so far. One of the missing things is the RPI 5 Runner, and I will need to update the libs once more.

File: binding/python/setup.py (hunk 21 above)
Comment (laves) [reply binding/python/setup.py:35]: I'm not clear on how this is handled via init - it still seems as though we are shipping both models, but only providing one as a default model in code. We should either ship a single model or offer the user the ability to choose between the two default models we ship.

File: README.md [hunk 32]
```diff
@@ -1,2 +1,208 @@
...
+- "{read|R IY D} this as {read|R EH D}, please."
+- "I {live|L IH V} in {Sevilla|S EH V IY Y AH}. We have great {live|L AY V} sports!"
+
+### Voices
+
+Orca can synthesize speech with different voices. The following table lists the available voices:
+
+|       Model name        | Sample rate (Hz) | 
+|:-----------------------:|:----------------:|
+| `orca_params_female.pv` |      22050       |
```
Comment (mrrostam) [inline README.md:61]: as discussed,  it might be better to put a link to these two files here

File: binding/python/setup.py (hunk 21 above)
Comment (mrrostam) [inline binding/python/setup.py:35]: I think you already discussed this with @laves 
we shouldn't include all files in this folder

File: binding/python/test_orca.py [hunk 33]
```diff
@@ -0,0 +1,172 @@
...
+        leopard = pvleopard.create(access_key=self.access_key)
+
+        for orca in self.orcas:
+            pcm = orca.synthesize(TEXT)
+            self.assertGreater(len(pcm), 0)
+
+            pcm = orca.synthesize(TEXT)
+
+            ground_truth = TEXT_NO_PUNCTUATION.split()
+            predicted, _ = leopard.process(pcm)
```
Comment (mrrostam) [inline binding/python/test_orca.py:74]: Nice! The good thing about this test is that we can easily implement it in other SDKs so long as we have access to Leopard

File: demo/c/test/test_orca_c.py [hunk 34]
```diff
@@ -0,0 +1,76 @@
...
+#
+
+import os.path
+import subprocess
+import sys
+import unittest
+
+from test_util import get_model_paths
+
+TEXT = "It doesn't matter how slowly you go, as long as you do not stop!"
```
Comment (mrrostam) [inline demo/c/test/test_orca_c.py:19]: It doesn't need to happen now, but I think it would be much better and easier if we remove the test data and place it in a [JSON file](https://github.com/Picovoice/rhino/blob/master/resources/.test/test_data.json). This way, we won't have to include this string and its corresponding WER in each SDK.

File: lib/ios/PvOrca.xcframework/ios-arm64_x86_64-simulator/PvOrca.framework/Headers/picovoice.h [hunk 35]
```diff
@@ -0,0 +1,85 @@
...
+extern "C" {
+
+#endif
+
+#define PV_API __attribute__((visibility("default")))
+
+/**
+ * Audio sample rate accepted by Picovoice.
+ */
+PV_API int32_t pv_sample_rate(void);
```
Comment (mrrostam) [inline lib/ios/PvOrca.xcframework/ios-arm64_x86_64-simulator/PvOrca.framework/Headers/picovoice.h:28]: remove

**Review by mrrostam — APPROVED** [review]

File: binding/python/setup.py (hunk 21 above)
Comment (bejager) [reply binding/python/setup.py:42]: Decided to include the female voice as default :)

File: demo/c/test/test_orca_c.py (hunk 34 above)
Comment (bejager) [reply demo/c/test/test_orca_c.py]: Moved the test data to the resource folder

File: binding/python/_util.py (hunk 20 above)
Comment (bejager) [reply binding/python/_util.py]: As discussed again, we ship only one model (the female one) as default..

**Comment by bejager** [comment]
This is good to go from my side

**Review by laves — APPROVED** [review]
Let Orca hit the seas!

**Review by ksyeo1010 — APPROVED** [review]
